frontend/src/editor/editor.tsx

```diff
@@ -18,6 +18,135 @@ import { OnChangePlugin } from '@lexical/react/LexicalOnChangePlugin';
 
 import classes from './editor.module.css';
 
+
+function $getDocContext(): DocContext {
+  // Initialize default empty context
+  const docContext: DocContext = {
```

Comment:
we keep repeating this. Perhaps we want a utility function `makeEmptyDocContext`. (I'm wary of making it an exported object because we'll accidentally modify it.)